### ISSUE TITLE
Envelope by Sketch throws exception thrown on dev.

### ISSUE THREAD

**Issue by ikeough** [hypar-io/BuildingBlocks#39]
Reported by @cdriesler 
```
Object reference not set to an instance of an object.: NullReferenceException
   at EnvelopeBySketch.EnvelopeBySketch.Execute(Dictionary`2 inputModels, EnvelopeBySketchInputs input) in /Users/ikeough/Documents/Hypar/BuildingBlocks/Envelope/EnvelopeBySketch/src/EnvelopeBySketch.cs:line 37
   at Hypar.Functions.Execution.InvocationWrapper`2.InvokeAsync(TInput input)
   at EnvelopeBySketch.Function.Handler(EnvelopeBySketchInputs args, ILambdaContext context) in /Users/ikeough/Documents/Hypar/BuildingBlocks/Envelope/EnvelopeBySketch/src/Function.g.cs:line 45
   at lambda_method(Closure , Stream , Stream , LambdaContextInternal )
```

**Comment by ikeough** [comment]
Required fix for #38.

**Comment by ikeough** [comment]
This is the result of the switch to `input_schema` where `SetbackDepth` is now only available if `UseSetbacks` is set to true.

**Comment by ikeough** [comment]
Originally I couldn't re-create @cdriesler's issue, using a workflow that already had this function placed. So I deleted the function and placed it again and got this:
![image](https://user-images.githubusercontent.com/1139788/105917926-f5272700-5fe7-11eb-8c90-7760d77b31e8.png)
The error was the same as above. The error above points to the `input.Perimeter` being null, which makes sense because the polygon has not been defined yet. What doesn't make sense is that previously we did not execute before all required inputs were satisfied. It seems like we are executing immediately, even before the UI for the function instance is loaded. This feels related to https://github.com/hypar-io/Explore/issues/1059.
